GitUI/Avatars/FileSystemAvatarCache.cs

```diff
@@ -138,21 +141,40 @@ public async Task ClearCacheAsync()
 
             if (_fileSystem.Directory.Exists(cachePath))
             {
-                await Task.Run(
-                    () =>
+                try
+                {
+                    string[] files = _fileSystem.Directory.GetFiles(cachePath);
+                    if (files.Length != 0)
                     {
-                        foreach (string file in _fileSystem.Directory.GetFiles(cachePath))
-                        {
-                            try
-                            {
-                                _fileSystem.File.Delete(file);
-                            }
-                            catch
+                        await Task.Run(
+                            async () =>
                             {
-                                // do nothing
-                            }
-                        }
-                    });
+                                foreach (string file in files)
+                                {
+                                    try
+                                    {
+                                        _fileSystem.File.Delete(file);
+                                    }
+                                    catch
+                                    {
+                                        // do nothing
+                                    }
+                                }
+
+                                if (CacheCleared != null)
+                                {
+                                    await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                                    CacheCleared.Invoke(this, EventArgs.Empty);
+                                }
+                            });
+                        return;
```

Comment:
* `return` is superfluous` because `CacheCleared` shall always be invoked
* `if` and `SwitchToMainThread` should be moved down

Reply:
>return is superfluousbecauseCacheCleared` shall always be invoked

The funny think is that while testing, I discovered that `CacheCleared` is always null because never used (event subscription is done on `MultiCacheCleaner`)